internal/prober/prober.go

```diff
@@ -152,13 +153,34 @@ func (p *Prober) probeAPIServer(shootClient kubernetes.Interface) error {
 }
 
 func (p *Prober) probeNodeLeases(shootClient kubernetes.Interface) ([]coordinationv1.Lease, error) {
-	nodeLeases, err := shootClient.CoordinationV1().Leases(nodeLeaseNamespace).List(p.ctx, metav1.ListOptions{})
+	nodes, err := shootClient.CoreV1().Nodes().List(p.ctx, metav1.ListOptions{})
+	if err != nil {
+		p.setBackOffIfThrottlingError(err)
+		p.l.Error(err, "Failed to list nodes, will retry probe")
+		return nil, err
+	}
+
+	nodeNames := sets.New[string]()
+	for _, node := range nodes.Items {
+		nodeNames.Insert(node.Name)
+	}
+
+	leases, err := shootClient.CoordinationV1().Leases(nodeLeaseNamespace).List(p.ctx, metav1.ListOptions{})
 	if err != nil {
 		p.setBackOffIfThrottlingError(err)
 		p.l.Error(err, "Failed to list leases, will retry probe")
 		return nil, err
 	}
-	return nodeLeases.Items, err
+
+	var out []coordinationv1.Lease
```

Comment:
Can you rename `out` to `nodeLeases`?

Reply:
Since `leases` is already defined. I would propose that you just rename this to `filteredLeases`